x-pack/plugins/spaces/server/spaces_service/spaces_service.ts

```diff
@@ -68,14 +68,18 @@ export class SpacesService {
       return spaceId;
     };
 
+    const internalRepositoryPromise = getStartServices().then(([coreStart]) =>
+      coreStart.savedObjects.createInternalRepository(['space'])
+    );
+
     const getScopedClient = async (request: KibanaRequest) => {
       const [coreStart] = await getStartServices();
+      const internalRepository = await internalRepositoryPromise;
```

Comment:
`createScopedRepository` already creates a new repository for every request, so after the other improvements it's probably not that critical that we re-use this repository instead of creating a new one. But since this is on the hot path it felt like it doesn't hurt to keep this minor improvement around.